### PR TITLE
First run at ECS boto.

### PR DESCRIPTION
**Pull request by sapiensantiquus**

### CODE REVIEW

File: aws_boto3/ecs/__init__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    cluster_arn = response['cluster']['clusterArn']
+    service_response = None
+    if 'services' in cluster:
+        services = cluster['services']
+        response['services'] = []
+        for service in services:
+            service['service_definition']['cluster'] = cluster_arn
+            service_response = ecs_ensure_service(service, region)
+            response['services'].append(service_response)
+    print response
```
Comment (meganlkm) [inline aws_boto3/ecs/__init__.py]: use a logger rather than print

File: aws_boto3/ecs/__init__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from aws_boto3.common import get_client
+from aws_boto3.ecs.services import ecs_ensure_service
+from aws_boto3.ecs.tasks import ecs_ensure_register_task
+
+
+
+def ecs_ensure_cluster(cluster, region=None):
+    client = get_client('ecs', region=region)
+    cluster_definition = cluster['cluster_definition']
+    response = client.create_cluster(**cluster_definition)
```
Comment (meganlkm) [inline aws_boto3/ecs/__init__.py]: what if the cluster already exists? should wrap this in a try block

File: aws_boto3/ecs/services.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from aws_boto3.ecs.tasks import ecs_ensure_register_task
+from aws_boto3.common import get_client
+
+def ecs_ensure_service(service,region=None):
```
Comment (meganlkm) [inline aws_boto3/ecs/services.py]: pep8 - space after the comma and 2 lines before the function definition

File: aws_boto3/ecs/tasks.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from aws_boto3.common import get_client
+
+def ecs_ensure_register_task(task, region=None):
+    client=get_client('ecs', region=region)
```
Comment (meganlkm) [inline aws_boto3/ecs/tasks.py]: pep8 - spaces before and after `=`

File: aws_boto3/ecs/tasks.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from aws_boto3.common import get_client
+
```
Comment (meganlkm) [inline aws_boto3/ecs/tasks.py]: pep8 - 2 lines before the function definition

File: aws_boto3/ecs/__init__.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+from aws_boto3.common import get_client
+from aws_boto3.ecs.services import ecs_ensure_service
+from aws_boto3.ecs.tasks import ecs_ensure_register_task
+
+
+
+def ecs_ensure_cluster(cluster, region=None):
```
Comment (meganlkm) [inline aws_boto3/ecs/__init__.py]: pep8 - only 2 lines before the function definition

File: aws_boto3/ecs/services.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    task_response = None
+    if 'task' in service:
+        task = service['task']
+        task['cluster_arn'] = service['service_definition']['cluster']
+        task['service_name'] = service['service_definition']['serviceName']
+        task_response = ecs_ensure_register_task(task,region)
```
Comment (meganlkm) [inline aws_boto3/ecs/services.py]: pep8 - space after the comma

File: aws_boto3/ecs/__init__.py (hunk 2 above)
Comment (sapiensantiquus) [reply aws_boto3/ecs/__init__.py]: i was planning on just letting it throw an error.

Comment (sapiensantiquus) [reply aws_boto3/ecs/__init__.py]: i think making ECS idempotent with boto is not going to be fun.

File: aws_boto3/ecs/__init__.py (hunk 1 above)
Comment (sapiensantiquus) [reply aws_boto3/ecs/__init__.py]: sorry, that is a debug line, i should just remove it.

File: aws_boto3/ecs/__init__.py (hunk 2 above)
Comment (sapiensantiquus) [reply aws_boto3/ecs/__init__.py]: My question would be: What would you like to do with the caught error?

**Pull request merged**